lib/gears/surface.lua

```diff
@@ -283,6 +284,39 @@ function surface.widget_to_surface(widget, width, height, format)
     return img, run_in_hierarchy(widget, cr, width, height)
 end
 
+--- Crop a surface to a given ratio
+-- This creates a copy of the surface, as surfaces seem to not be
+-- resizable after creation, so you have to use the functions return
+-- value. The resulting surface is in the center of the original surface
+-- @param surf surface the Cairo surface to crop
+-- @param ratio number the ratio the image should be cropped to (widht/height)
+-- @return the a cropped copy of the input surface
+function surface.crop_surface(surf, ratio)
```

Comment:
This is a very generic (and thereby "valuable") function name, but the signature and implementation only supports a very small subset of what that feature could be.
My initial assumption for a "crop image" API would have been to allow arbitrary values to be cut on each side, e.g. `crop(img, { left = 15, right = 27, top = 0, bottom = 3})`.

Even if arbitrary cropping is not something that's needed/implemented right now, the signature should at least be changed to allow for that in the future without breaking changes.

Reply:
yup using `args` table instead would be better in long-term indeed

Reply:
the function i submitted was only meant to be used with a fixed ratio.
I am gonna rename this and work on a general crop function as well, as i am already on it

Reply:
i meant that you could just rename `ratio` to `args` and use `args.ratio` in the code, so it would be easier to extend the func without making the API ridiculous in the future

Reply:
With an args table, it should be the only parameter (because people got used to the fact that functions can be called like `foo { bar = 1, baz = "baz" }`). Then you'd use that to allow for both `crop({ surface = img, ratio = 3/5 })` and `crop({ surface = img, left = 3, right = 3})`.

Separated and renamed to `crop_surface_to_ratio` would be fine, too, though.

> Wouldn't it make more sense out of a users perspective if you assigned a target width and height that the image then gets cropped to?

That would still not cover all use cases. With just a width and height, the anchor would still be fixed (like your current version also fixes it to always crop towards the center).
You could do `width` + `height` + `x` + `y`, but that would usually lead to more calculations required to get those values than the one for each edge (not corner).